BtConnectorLib/btconnectorlib2/src/main/java/org/thaliproject/p2p/btconnectorlib/BTConnector.java

```diff
@@ -177,24 +182,25 @@ private void startServices() {
 
         WifiP2pManager.Channel channel = null;
         WifiP2pManager p2p = null;
-        if (mWifiBase != null) {
-            channel = mWifiBase.GetWifiChannel();
-            p2p = mWifiBase.GetWifiP2pManager();
+        WifiBase tmp = mWifiBase;
+        if (tmp != null) {
+            channel = tmp.GetWifiChannel();
+            p2p = tmp.GetWifiP2pManager();
         }
 
         if (channel != null && p2p != null) {
             print_line("", "Starting services address: " + mInstanceString + ", " + ConSettings);
-
-            mBTConnector_Discovery = new BTConnector_Discovery(channel,p2p,this.context,this,ConSettings.SERVICE_TYPE,mInstanceString);
-            mBTConnector_Discovery.Start();
+            BTConnector_Discovery tmpDisc= new BTConnector_Discovery(channel,p2p,this.context,this,ConSettings.SERVICE_TYPE,mInstanceString);
+            tmpDisc.Start();
+            mBTConnector_Discovery = tmpDisc;
         }
     }
 
     private  void stopServices() {
```

Comment:
And shouldn't there be a check that startServices was called before stopServices?